From 12f5166d46db1459455b33b32c01b1fde5c73701 Mon Sep 17 00:00:00 2001
From: Nikos Mavrogiannopoulos <nmav@redhat.com>
Date: Mon, 22 Aug 2016 08:18:15 +0200
Subject: [PATCH] gnutls-cli: print 'Handshake was completed'

The change of moving the handshake process as part of the socket
establishment, prevented the text 'Handshake was completed' from
being printed as part of a successful handshake. That message was
used by applications like gnus which use gnutls-cli. This patch
reverts that change and prints that message on successful handshakes.
---
 src/cli.c    | 2 ++
 src/socket.c | 3 +--
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/cli.c b/src/cli.c
index 9248222..0913898 100644
--- a/src/cli.c
+++ b/src/cli.c
@@ -1218,6 +1218,8 @@ int main(int argc, char **argv)
 	socket_open(&hd, hostname, service, OPT_ARG(STARTTLS_PROTO), socket_flags, CONNECT_MSG, NULL);
 	hd.verbose = verbose;
 
+	printf("- Handshake was completed\n");
+
 	if (resume != 0)
 		if (try_resume(&hd))
 			return 1;
diff --git a/src/socket.c b/src/socket.c
index aae9ee7..5507c71 100644
--- a/src/socket.c
+++ b/src/socket.c
@@ -465,8 +465,7 @@ socket_open(socket_st * hd, const char *hostname, const char *service,
 			else if (err < 0) {
 				fprintf(stderr, "*** handshake has failed: %s\n", gnutls_strerror(err));
 				exit(1);
-			} else if (hd->verbose)
-				printf("- Handshake was completed\n");
+			}
 		}
 
 		break;
-- 
2.9.3

